### ISSUE TITLE
Last version Build error: could not load assembly 'Plugin.Permissions.Abstractions, Version=2.2.1.0

### ISSUE THREAD

**Issue by VahidShir** [jamesmontemagno/PermissionsPlugin#98]
I installed last version(3.0.0.8) successfully on both shared and Android projects. But when I wanna build I get this error:
`error : Exception while loading assemblies: System.IO.FileNotFoundException: Could not load assembly 'Plugin.Permissions.Abstractions, Version=2.2.1.0, Culture=neutral, PublicKeyToken='. Perhaps it doesn't exist in the Mono for Android profile?`
`error : File name: 'Plugin.Permissions.Abstractions.dll`

However if I install a previous version such as (2.2.1) the project would build successfully.

My config:
- Visual Studio: 15.6.6
- .NetStandard 2.0
- Xamarin.Forms: 2.5.1.527436
- Target Android Version: Android 8.1 (API 27)
- Compile Android version: Android 8.1

**Comment by jamesmontemagno** [comment]
Do a full clean and re-build as I no longer load that assembly it is safe to remove.

**Comment by perfectsine** [comment]
This appears to be working locally on Xamarin, but when I upload to Microsoft Appcenter, it's failing with the same error for me.  App center does clean builds with every build, not sure what needs to happen here.   I'm having to revert as well.

**Comment by VahidShir** [comment]
I already did clean the whole solution as well as deleting obj/bin folders but this didn't resolve the issue.
But surprisingly in Release it built successfully and after switching back to Debug mode the build error went away! It doesn't make any sense to me!

**Comment by jamesmontemagno** [comment]
Delete the bin / obj

**Comment by kyluke** [comment]
That doesn't seem to work for me.

Tried:
 - Deleting bin / obj folders
 - Clean & Debug Rebuild
---
 - Deleting bin / obj folders
 - Clean & Release rebuild

I get the same error message as above. Downgrading to 2.2.1 for now in order to continue development.

**Comment by VahidShir** [comment]
Another point that might help is as I have another package named "_Xam.Plugin.Geolocator_" which has dependency on _PermissionsPlugin_, I updated that as well.

**Comment by jamesmontemagno** [comment]
I have rolled back this updated for now.

**Comment by pamela032709** [comment]
I got the same exact issue. and I did all those steps. 
any other thing I could do?

**Comment by K232** [comment]
I'm using latest 3.0.0.12 and also got (nearly) the same error, except for me it's search for Version 1.2.1.0. Tried deleting bin / obj and rebuild, no changes. Downgrading to 2.2.1 fixed the problem.

_/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(2,2): Error: Exception while loading assemblies: System.IO.FileNotFoundException: Could not load assembly 'Plugin.Permissions.Abstractions, Version=1.2.1.0, Culture=neutral, PublicKeyToken='. Perhaps it doesn't exist in the Mono for Android profile?
File name: 'Plugin.Permissions.Abstractions.dll'
  at Java.Interop.Tools.Cecil.DirectoryAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference reference, Mono.Cecil.ReaderParameters parameters) [0x0009a] in /Users/builder/data/lanes/5945/342b2ce9/source/monodroid/external/xamarin-android/external/Java.Interop/src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs:229 
  at Java.Interop.Tools.Cecil.DirectoryAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference reference) [0x00000] in /Users/builder/data/lanes/5945/342b2ce9/source/monodroid/external/xamarin-android/external/Java.Interop/src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs:179 
  at Xamarin.Android.Tasks.ResolveAssemblies.AddAssemblyReferences (Java.Interop.Tools.Cecil.DirectoryAssemblyResolver resolver, System.Collections.Generic.ICollection`1[T] assemblies, Mono.Cecil.AssemblyDefinition assembly, System.Boolean topLevel) [0x0014a] in <fdfe8f54615a4e2ab24c72dc90da5c64>:0 
  at Xamarin.Android.Tasks.ResolveAssemblies.Execute (Java.Interop.Tools.Cecil.DirectoryAssemblyResolver resolver) [0x00237] in <fdfe8f54615a4e2ab24c72dc90da5c64>:0_

**Comment by pamela032709** [comment]
You are so right. Thanks, thanks. I did downgrade to 2.2.1 and it worked. so I guess that the latest version still not compatible with xam.plugin.media and geolocator. I don't know why. but is like you have to play around with the versions to see which one builds.

**Comment by Genfood** [comment]
I had the same problem with the version 3.0.0.12. But only under Android with the Target Framework 6. Changing the Target Framework to 8 could probably solve the problem as well.

**Comment by angelieri** [comment]
Just to document de issue, I'm facing the same problem with 3.0.0.12.
A need this update because Geolocator.

**Comment by emicarito** [comment]
angelieri solution works for me thanks

**Comment by jrgsc** [comment]
the solution by angelieri worked for me. I simply downgraded Geolocator to 4.2.0 and the problem went away.

**Comment by fwermelinger** [comment]
I am not using Geolocator but got the same error when building for iOS. Android worked fine. The solution by worked for me as well.

**Comment by abreucruz** [comment]
I downgraded the android version to 2.2.1.0 only and it works fine

**Comment by faceoffers28** [comment]
I had a similar issue after upgrading to 4.5.0.6 of the Geolocator Plugin. To fix this issue, I had to upgrade from version 3.0.1 of the Xam.Plugin.Media Plugin to version 4.0.1.5. It appears that all these Xam.Plugin packages really need to be upgraded together. As @jamesmontemagno indicated, I did a clean, rebuild and deleted the Debug folders in the obj and bin folders. This did not work. The fix for me was to update the Xam.Plugin.Media package.
